core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptVersionConfiguration.java

```diff
@@ -145,26 +195,75 @@ public void insert(ScriptVersion scriptVersion) {
         getMetadataRepository().executeUpdate(getInsertStatement(scriptVersion));
     }
 
-    private String deleteStatement(ScriptVersionKey scriptVersionKey) {
-        return "DELETE FROM " + getMetadataRepository().getTableNameByLabel("ScriptVersions") +
+    @Override
+    public void update(ScriptVersion scriptVersion) {
+        LOGGER.trace(MessageFormat.format("Updating ScriptVersion {0}-{1}.", scriptVersion.getScriptId(), scriptVersion.getNumber()));
+        if (!exists(scriptVersion)) {
+            throw new MetadataDoesNotExistException(scriptVersion);
+        }
+        getMetadataRepository().executeUpdate(getUpdateStatement(scriptVersion));
+    }
+
+    public void restoreDeletedScriptVersion(ScriptVersion scriptVersion){
+        LOGGER.trace(MessageFormat.format("Restoring deleted ScriptVersion {0}", scriptVersion.getMetadataKey().toString()));
+        if (exists(new ScriptVersionKey(scriptVersion.getMetadataKey().getScriptKey()))){
```

Comment:
scriptVersion.getMetadataKey().getScriptKey()) will contain the deletedAt property. You'll have to check if a script version exists with only scriptId & scriptVersion (so NOT checking deletedAt)

Reply:
Already we are doing same @BerrevoetsRobbe